src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -100,56 +100,6 @@ void RealmCoordinator::create_sync_session()
     if (m_sync_session)
         return;
 
-    // Create the fresh config copy here. A copy is needed so that a self reference doesn't keep the sync config alive
```

Comment:
I pulled this code into sync_session.cpp rather than splitting it awkwardly between here and there, and made it just use DB and SyncSession directly rather than go through RealmCoordinator as it wasn't simplified by using RealmCoordinator.